Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/RTTI/RTTI.h>
+#include <AzFramework/Entity/EntityContext.h>
+#include <AzToolsFramework/Prefab/Instance/InstanceDomGeneratorInterface.h>
+#include <AzToolsFramework/Prefab/PrefabDomTypes.h>
+
+namespace AzToolsFramework
+{
+    namespace Prefab
+    {
+        class Instance;
+        class PrefabFocusInterface;
+        class PrefabSystemComponentInterface;
+
+        class InstanceDomGenerator
+            : public InstanceDomGeneratorInterface
+        {
+        public:
+            AZ_RTTI(InstanceDomGenerator, "{07E23525-1D91-41AA-B85F-136360BD1938}", InstanceDomGeneratorInterface);
+            AZ_CLASS_ALLOCATOR(InstanceDomGenerator, AZ::SystemAllocator, 0);
+
+            bool GenerateInstanceDom(const Instance* instance, PrefabDom& instanceDom) override;
+
+            void RegisterInstanceDomGeneratorInterface();
+            void UnregisterInstanceDomGeneratorInterface();
```

Comment:
nit: I'd just call these "Initialize" and "Uninitialize" since the function does not just register the interface.